### PR TITLE
Port ComparisonAcceptanceTest to TCK

### PR DESCRIPTION
**Pull request by Mats-SX**

### CODE REVIEW

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    And having executed: CREATE INDEX ON :Product(unitsInStock)
+    And having executed:
+      """
+      CREATE (:Product {unitsInStock: 8})
+      CREATE (:Product {unitsInStock: 12})
+      """
+    When executing query:
+      """
+      MATCH (p:Product)
+      USING INDEX p:Product(unitsInStock)
```
Comment (Mats-SX) [inline tck/features/ComparisonOperatorAcceptance.feature]: Index hints are not part of openCypher.

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
```
Comment (Mats-SX) [inline tck/features/ComparisonOperatorAcceptance.feature]: Index hints are not part of openCypher.

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  Scenario: Handling numerical ranges 1
+    Given an empty graph
+    And having executed:
+      """
+      UNWIND [1, 2, 3] AS i
+      CREATE ({value: i})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 1 < n.value < 3
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: I would suggest putting the `WHERE` clause on a new line (Cypher style)

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+  Scenario: Handling numerical ranges 2
+    Given an empty graph
+    And having executed:
+      """
+      UNWIND [1, 2, 3] AS i
+      CREATE ({value: i})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 1 < n.value <= 3
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: The `WHERE` clause ought to go on a new line

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+  Scenario: Handling numerical ranges 3
+    Given an empty graph
+    And having executed:
+      """
+      UNWIND [1, 2, 3] AS i
+      CREATE ({value: i})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 1 <= n.value < 3
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: The `WHERE` clause ought to go on a new line

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+  Scenario: Handling numerical ranges 4
+    Given an empty graph
+    And having executed:
+      """
+      UNWIND [1, 2, 3] AS i
+      CREATE ({value: i})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 1 <= n.value <= 3
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: The `WHERE` clause ought to go on a new line

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+  Scenario: Handling string ranges 1
+    Given an empty graph
+    And having executed:
+      """
+      UNWIND ['a', 'b', 'c'] AS c
+      CREATE ({value: c})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 'a' < n.value < 'c'
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: The `WHERE` clause ought to go on a new line

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+  Scenario: Handling string ranges 2
+    Given an empty graph
+    And having executed:
+      """
+      UNWIND ['a', 'b', 'c'] AS c
+      CREATE ({value: c})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 'a' < n.value <= 'c'
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: The `WHERE` clause ought to go on a new line (plus, all the queries containing `WHERE` below this one)

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+  Scenario: Handling string ranges 3
+    Given an empty graph
+    And having executed:
+      """
+      UNWIND ['a', 'b', 'c'] AS c
+      CREATE ({value: c})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 'a' <= n.value < 'c'
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: `WHERE` ought to go on a new line

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+  Scenario: Handling string ranges 4
+    Given an empty graph
+    And having executed:
+      """
+      UNWIND ['a', 'b', 'c'] AS c
+      CREATE ({value: c})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 'a' <= n.value <= 'c'
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: `WHERE` ought to go on a new line

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+
+  Scenario: Handling empty range
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ({value: 3})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE 10 < n.value <= 3
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: `WHERE` ought to go on a new line

File: tck/features/ComparisonOperatorAcceptance.feature [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+      CREATE (a:A {prop1: 3, prop2: 4})
+      CREATE (b:B {prop1: 4, prop2: 5})
+      CREATE (c:C {prop1: 4, prop2: 4})
+      CREATE (a)-[:R]->(b)
+      CREATE (b)-[:R]->(c)
+      CREATE (c)-[:R]->(a)
+      """
+    When executing query:
+      """
+      MATCH (n)-->(m) WHERE n.prop1 < m.prop1 = n.prop2 <> m.prop2
```
Comment (petraselmer) [inline tck/features/ComparisonOperatorAcceptance.feature]: `WHERE` ought to go on a new line

**Comment by Mats-SX** [comment]
@petraselmer Corrected comments, thank you!

**Pull request merged**